itwin-platform-access/imodels-access-backend/src/BackendIModelsAccess.ts

```diff
@@ -371,17 +376,18 @@ export class BackendIModelsAccess implements BackendHubAccess {
     if (locks.length === 0)
       return;
 
-    const lock: Lock = locks[0];
-    this.setLockLevelToNone(lock.lockedObjects);
+    for await (const locksPage of locks){
```

Comment:
Instead of using toArray above, better load page by page and instantly release them. This way we will save memory.

Reply:
This does not work with current paging implementation. As after releasing first page the next 100 are different. Basically it skips every second page. We need API to support continuation tokens first.

Reply:
Maybe then we can always query first 100 and release only them?

Reply:
fixed with https://github.com/iTwin/imodels-clients/pull/181/commits/59f3c709a2860ccddedcc7b73d4798cc20c9c842